plugins/plugin_utils/base/ipaddress_utils.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The utils file for all netaddr tests
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+import json
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils.six import ensure_text
+from functools import wraps
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+try:
+    import ipaddress
+    HAS_IPADDRESS = True
+except ImportError:
+    HAS_IPADDRESS = False
+
+
+def ip_network(ip):
+    """ PY2 compat shim, PY2 requires unicode
+    """
+
+    if not HAS_IPADDRESS:
+        msg = "Missing required library 'ipaddress'"
+        raise AnsibleError(msg)
```

Comment:
This can be replaced by `missing_required_libs` https://github.com/ansible-collections/ansible.netcommon/blob/main/plugins/connection/netconf.py#L323